src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -45,20 +39,32 @@
          auth-handler
          puppet-version)
        {:normalize-request-uri true})
-     (when (not= ca-crl-file host-crl-file)
-       (watch-protocol/add-watch-dir! watcher
-                                      (fs/parent ca-crl-file)
-                                      {:recursive true})
-       (watch-protocol/add-callback!
-        watcher
-        (fn [events]
-          (when (some #(and (:changed-path %)
-                            (= (.getCanonicalPath (:changed-path %))
-                               ca-crl-file))
-                      events)
-            (ca/retrieve-ca-crl! ca-crl-file host-crl-file)))))
-     (assoc context :auth-handler auth-handler
-                    :watcher watcher)))
+     (-> context
+         (assoc :auth-handler auth-handler)
+         ((fn [context]
```

Comment:
I don't think this necessarily adds to the readability of what's happening, and while using fp, I don't think it's more idiomatic than say adding
```clj
context' (assoc :auth-handler auth-handler)
```
in the [above `let`](https://github.com/puppetlabs/puppetserver/pull/1475/files#diff-3e475421e7cb19ba1f757a133f75e650R27), and then having just the `if-let` (without the anonymous fn) operate on `context'` rather than `context`.